db/data_migrations/20200108105400_create_backend_apis.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+require 'progress_counter'
+
+class CreateBackendApis < ActiveRecord::DataMigration
```

Comment:
Not sure if we should still be doing this. Perhaps this migration is outdated.

We originally decided to create one Backend per existing service because of Saas and on-prem customers migrating to 2.7. However, 2.7 also allows a Product to have no Backend at all. So, in on-prem only, where there's a rake task that triggers this migration on every deploy, system will always create Backend records for every Product missing a Backend. This smells like a bug to be, which this PR will also fix by turning this migration again into... well... a migration, rather than a rake task that triggers on every deploy.

Reply:
Valid for on-prem upgrades from 2.6 to 2.7. Plus we already ran it on saas. So, no longer needed?

Reply:
Indeed, @thomasmaas. That's why it's still a draft :) The reason why I decided to move this rake task only was to provide an example of how data migrations would be written if we introduce the gem.

Now our options for this migration specifically are:
1. remove it completely (together with its original rake task)
2. skip it in the code with an ugly `return` in the beginning as we did a couple times in the past (meh)
3. inject '20200108105400' into the `data_migrations` table to prevent it from running again

Reply:
I'd pick 1 but something tells me you would pick 3

Reply:
Not at all. I'd pick one too. No code is better than any code.

Reply:
Extracted to a separate commit so we can remove it later. But first I'd like to get some feedback from the team about this is way to handle the data migrations.

Reply:
I also prefer the `option 1` for this. We only needed that for 2.7, so in that point in time the code will always be there. But I wouldn't keep it and change it for it not to run XDD I find it pointless and I am definitely not a big fan of keeping code just for the sake of it because it is only useful to ask for problems.

![kill it with fire](https://media.giphy.com/media/wvFrJa4aWKuvm/giphy.gif)

Reply:
I prefer the 1st also

Reply:
Just remove it. If I understand it well like migration, it is based on versioning so next new install will not run those data migrations. Past is past: option 1